### PR TITLE
Add binding interface to discover_ssdp

### LINKED ISSUE #58: discover_ssdp is not accepting interface to bind to
def discover_ssdp(port=1900, ttl=2, response_time=3)

should accept also interface or ip addr so we can define which interface to use to multicast via...

### PR DESCRIPTION
**Pull request by glimchb**
Fix #58
@mraineri please review

### CODE REVIEW

**Comment by mraineri** [comment]
Is this `+'\0'` really necessary? Other than that, everything else looks good to me.

**Comment by glimchb** [comment]
> Is this `+'\0'` really necessary? Other than that, everything else looks good to me.

Seems like we have to... 
The other way to do it is something like this if you prefer it: `s.setsockopt(socket.SOL_SOCKET,IN.SO_BINDTODEVICE,struct.pack("%ds" % (len("eth0")+1,), "eth0"))`

**Comment by mraineri** [comment]
Oh yuck, I see that now. I forgot that the value parameter for setsockopt either takes an integer or a buffer with a C-like structure.

File: src/redfish/discovery/discovery.py [hunk 1]
```diff
@@ -72,6 +72,8 @@ def discover_ssdp(port=1900, ttl=2, response_time=3):
     sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
     sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
     sock.setsockopt(socket.IPPROTO_IP, socket.IP_MULTICAST_TTL, ttl)
+    if iface:
+        sock.setsockopt(socket.SOL_SOCKET, socket.SO_BINDTODEVICE, iface+'\0')
```
Comment (MoshiBin) [inline src/redfish/discovery/discovery.py]: We may have to pass `bytes(iface + "\0")` to be compatible with python3

Comment (glimchb) [reply src/redfish/discovery/discovery.py]: @MoshiBin not sure you are right...
```
[root@localhost ~]# python2 -c "print bytes('eth0'+'\0')"
eth0
```
vs
```
[root@localhost ~]# python3 -c "print bytes('eth0'+'\0')"
  File "<string>", line 1
    print bytes('eth0'+'\0')
              ^
SyntaxError: invalid syntax
```

**Comment by mraineri** [comment]
8/15: Approved, pending the change to "bytes(b'eth0'+b'\0')" (@mraineri will manage this when performing the release)

**Pull request merged**